Mathlib/Tactic/HashCommandLinter.lean

```diff
@@ -0,0 +1,78 @@
+/-
+Copyright (c) 2024 Damiano Testa. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Damiano Testa
+-/
+
+import Lean.Elab.Command
+import Lean.Linter.Util
+import Batteries.Lean.HashSet
+
+/-!
+#  `#`-command linter
+
+The `#`-command linter produces a warning when a command starting with `#` is used *and*
+* either the command emits no message;
+* or `warningAsError` is set to `true`.
+-/
```

Comment:
Long-term, I'd really appreciate if all Lean and mathlib linters had an explanation "why is this [code pattern what we're linting] bad". For instance, in Rust, the clippy linter has these ([example](https://rust-lang.github.io/rust-clippy/master/index.html#/absolute_paths), and I found them *really* helpful.

Not strictly required, but: do you mind adding one or two sentences on why we want this?